makefiles/Base.toml

```diff
@@ -7,26 +7,61 @@ CARGO_PROFILE = "release"
 [tasks.set-image-tag]
 env = { IMAGE_TAG = "${CARGO_MAKE_GIT_HEAD_LAST_COMMIT_HASH}-${CARGO_PROFILE}" }
 
-[env]
-FLUVIO_VERSION = { script = ["cat VERSION"] }
+################################################################################
+# Tasks for building targets matching the build host
+#
 
-[tasks.build-release.mac]
+[tasks.build-dev.mac]
+extend = "build-dev-mac"
+[tasks.build-dev.linux]
+extend = "build-dev-musl"
+
+[tasks.build-prod.mac]
+extend = "build-prod-mac"
+[tasks.build-prod.linux]
+extend = "build-prod-musl"
+
+
+################################################################################
+# Building for x86_64-apple-darwin can only be done on Mac
+#
+
+[tasks.build-dev-mac.mac]
+command = "cargo"
+args = ["build", "--bin", "fluvio", "--target", "x86_64-apple-darwin"]
+
+[tasks.build-prod-mac.mac]
+command = "cargo"
+args = ["build", "--bin", "fluvio", "--release", "--target=x86_64-apple-darwin"]
+
+# Fail apple-darwin build on linux
+[tasks.build-dev-mac.linux]
+script = "echo Cannot build for Mac from Linux && false"
+[tasks.build-prod-mac.linux]
+script = "echo Cannot build for Mac from Linux && false"
+
+
+################################################################################
+# Building linux-musl on Linux requires setting up musl-gcc toolchain
+#
+
+[tasks.build-dev-musl.linux]
+dependencies = ["setup-musl"]
+command = "cargo"
 args = [
     "build",
     "--bin", "fluvio",
-    "--release",
-    "--target=x86_64-apple-darwin",
-    "@@split(CARGO_MAKE_CARGO_BUILD_TEST_FLAGS, )"
+    "--target=x86_64-unknown-linux-musl",
 ]
 
-[tasks.build-release.linux]
+[tasks.build-prod-musl.linux]
```

Comment:
is there reason to renaming and changing this task? Since this is used by other action, they need to be tested as well

Reply:
Yes, there is a specific reason, which is that `build-release` is one of the magic "default tasks" of `cargo make`

- https://github.com/sagiegurari/cargo-make/blob/master/src/lib/descriptor/makefiles/rust.toml#L380

So when we add definitions to a task by the same name, we are not creating an original task, but rather we are editing the existing task, which is error prone and causes weird implicit effects in the task. I have found it much more effective to always create unique task names, hence `build-prod-musl`

Reply:
which weird effect?

Reply:
Just the fact that you would _think_ you are looking at the full task definition when in reality there are other things that get implicitly mixed in.